awsService/src/main/java/com/kovan/app/service/S3Service.java

```diff
@@ -0,0 +1,269 @@
+        } catch (S3Exception e) {
+            log.error("Error renaming bucket from {} to {}: {}", oldBucketName, newBucketName, e.getMessage());
+            return "Error renaming bucket: " + e.getMessage();
+        } catch (Exception e) {
+            log.error("Unexpected error renaming bucket: {}", e.getMessage());
+            return "Unexpected error renaming bucket: " + e.getMessage();
+        }
+    }
+    public String uploadFile(MultipartFile file) {
+
+        String fileTypeFolder = determineFolder(file.getOriginalFilename());
+        String path = fileTypeFolder + "/" + file.getOriginalFilename();
+        String uniqueId = randomUUID().toString();
+        service.saveDocument(Document.builder()
+                .id(uniqueId)
+                .fileName(path).build());
+        try {
+            log.info("Uploading file {} to bucket {}, key: {}", file.getOriginalFilename(), bucketName, path);
+            PutObjectRequest putObjectRequest = PutObjectRequest.builder()
+                    .bucket(bucketName)
+                    .key(path)
+                    .build();
+            s3Client.putObject(putObjectRequest,RequestBody.fromInputStream(file.getInputStream(), file.getSize()));
+        } catch (Exception e) {
+            log.error("Error uploading file to S3", e);
+            throw new FileException("File upload failed", e);
+        }
+        return uniqueId;
+    }
+
+    public String determineFolder(String fileName) {
+
+        String fileExtension = getFileExtension(fileName);
+        return switch (fileExtension.toLowerCase()) {
+            case "pdf" -> "pdfs";
+            case "doc", "docx" -> "docs";
+            case "jpg", "jpeg", "png" -> "images";
+            case "txt" -> "texts";
+            default -> "others";
+        };
+    }
+
+    private String getFileExtension(String fileName) {
+        if (isBlank(fileName) || !fileName.contains(".")) {
+            throw new FileException("Invalid file name: " + fileName);
+        }
+        return fileName.substring(fileName.lastIndexOf('.') + 1);
+    }
+
+    public byte[] downloadFile(String id) {
+
+        String fileName = service.findDocumentById(id).getFileName();
+
+        GetObjectRequest getObjectRequest = GetObjectRequest.builder()
+                .bucket(bucketName)
+                .key(fileName)
+                .build();
+
+        try (ResponseInputStream<GetObjectResponse> inputStream = s3Client.getObject(getObjectRequest);
+             ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+
+            byte[] buffer = new byte[1024];
+            int bytesRead;
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                outputStream.write(buffer, 0, bytesRead);
+            }
+
+            log.info("File with id {} downloaded successfully from bucket {}", id, bucketName);
+            return outputStream.toByteArray();
+        } catch (IOException e) {
```

Comment:
Improve logging will this method only throw IOException ? not any other ? try to imporve exception handling

Reply:
Improved exception handling.